hw2 is done

Ltaudato1 opened the pull request.

Ltaudato1 committed:
hw2 is done

Ltaudato1 committed:
Update GitHubClientIntegrationTest.java

Ltaudato1 committed:
Update GitHubClientIntegrationTest.java

Ltaudato1 committed:
hw3 is done

Ltaudato1 committed:
update test

damir2407 commented on scrapper/src/main/java/edu/java/configuration/ClientConfiguration.java:
```diff
@@ -0,0 +1,22 @@
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Configuration
+@Validated
```
А зачем тут Validated?

damir2407 commented on scrapper/src/main/java/edu/java/clients/StackOverflowClient.java:
```diff
@@ -0,0 +1,28 @@
+    private final WebClient webClient;
+
+    public StackOverflowClient(WebClient.Builder webClientBuilder, String baseUrl) {
+        this.webClient = webClientBuilder.baseUrl(baseUrl).build();
```
а че прям в самой конфигурации клиентов не внедрить эти WebClient'ы вместо билдеров? Давай сделаем так для обоих клиентов

damir2407 commented on scrapper/src/main/java/edu/java/configuration/ClientConfiguration.java:
```diff
@@ -0,0 +1,22 @@
+public class ClientConfiguration {
+    @Bean
+    public GitHubClient gitHubClient(WebClient.Builder webClientBuilder) {
+        return new GitHubClient(webClientBuilder, "https://api.github.com");
```
> При создании клиента обязательно должна быть возможность указать базовый URL. При этом если он не указывается, то должен использоваться URL по умолчанию.

Не нашел у тебя этого для обоих клиентов

damir2407 commented on scrapper/src/main/java/edu/java/clients/StackOverflowClient.java:
```diff
@@ -0,0 +1,28 @@
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Component
```
Ты же уже создал этих клиентов через @Bean в конф классе. Тут не нужна @Component, для гитхаба тоже

damir2407 commented on scrapper/src/main/java/edu/java/clients/GitHubClient.java:
```diff
@@ -0,0 +1,30 @@
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Component
+public class GitHubClient {
```
Давай как-то явно разделим по пакетам гитхаб и стаковерфлоу. Сложно читать клиентов и их дто

damir2407 commented on scrapper/src/test/java/edu/java/scrapper/GitHubClientIntegrationTest.java:
```diff
@@ -0,0 +1,48 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class GitHubClientIntegrationTest {
```
> Протестируйте клиентов при помощи библиотеки [WireMock](https://wiremock.org/) (зависимость уже добавлена).

Не нашел у тебя использование WireMock'а

Ltaudato1 committed:
hw4 is done

Ltaudato1 committed:
test fix

Ltaudato1 committed:
will be fixed in hw2

Ltaudato1 committed:
hw5 is done

Ltaudato1 committed:
hw6 is done

Ltaudato1 committed:
hw7 is done

Ltaudato1 committed:
hw8 is done

Ltaudato1 committed:
hw9

Ltaudato1 committed:
hw10

Ltaudato1 committed:
updated workflow

Ltaudato1 committed:
added jars

Ltaudato1 committed:
ACTUALLY added jars

Ltaudato1 committed:
updated docker-files

Ltaudato1 committed:
temporary fix

Ltaudato1 committed:
temporary fix v2

почему-то именно на гите докер билд вообще не проходит, хотя локально всё работает вообще отлично

Ltaudato1 committed:
refactor hw5

Ltaudato1 committed:
fix checkstyle

Ltaudato1 committed:
refactor hw7

Ltaudato1 committed:
refactor hw2

Ltaudato1 committed:
.

Vladislav-Kisliy reviewed (approved):
Тестов на базе WireMock так и не добавили. Зачту со снижением баллов

Ltaudato1 committed:
Merge pull request #11 from Ltaudato1/hw10

hw10

Ltaudato1 committed:
Merge pull request #10 from Ltaudato1/hw9

hw9

Ltaudato1 committed:
Merge pull request #9 from Ltaudato1/hw8

hw8 is done

Ltaudato1 committed:
Merge pull request #8 from Ltaudato1/hw7

hw7 is done

Ltaudato1 committed:
solving merge conflict pt.1

Ltaudato1 committed:
Merge branch 'hw6' of https://github.com/Ltaudato1/tinkoff_edu_spring into hw6

Ltaudato1 committed:
Merge branch 'hw6' into hw5

Ltaudato1 committed:
Merge branch 'hw3' into hw2
